core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -50,10 +50,11 @@ import org.apache.spark.util.{Clock, SystemClock, Utils}
 private[scheduler] class BlacklistTracker (
     private val listenerBus: LiveListenerBus,
     conf: SparkConf,
+    scheduler: Option[ExecutorAllocationClient],
     clock: Clock = new SystemClock()) extends Logging {
 
-  def this(sc: SparkContext) = {
-    this(sc.listenerBus, sc.conf)
+  def this(sc: SparkContext, scheduler: Option[ExecutorAllocationClient]) = {
```

Comment:
perhaps the variable here should be allocationClient or similar instead of scheduler since it wouldn't have to be a scheduler backend.